crm_automated_activity/models/crm_lead.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2021 Eder Brito
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import date, timedelta
+
+from odoo import api, models
+
+
+class CrmLead(models.Model):
+
+    _inherit = "crm.lead"
+
+    @api.model
+    def create(self, vals):
+
+        activity = self.env["mail.activity"]
+        crm_model = self.env["ir.model"].search([("model", "=", "crm.lead")])
+
+        res = super(CrmLead, self).create(vals)
+
+        if res.stage_id.create_automated_activity and any(
+            res.stage_id.automated_activity_ids
+        ):
+
+            for act in res.stage_id.automated_activity_ids:
+
+                deadline = date.today() + timedelta(days=act.days_deadline)
+
+                if act.apply_in in ["create", "create_write"]:
+
+                    activity.sudo().create(
+                        [
+                            {
+                                "res_id": res.id,
```

Comment:
Could you put the values retrievement in a separate function in order to easy inheritance ?